embedded-postgresql/src/main/java/com/playtika/test/postgresql/EmbeddedPostgreSQLBootstrapConfiguration.java

```diff
@@ -0,0 +1,97 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.playtika.test.postgresql;
+
+import java.util.LinkedHashMap;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.testcontainers.containers.GenericContainer;
+
+import static com.playtika.test.common.utils.ContainerUtils.containerLogsConsumer;
+import static com.playtika.test.postgresql.PostgreSQLProperties.BEAN_NAME_EMBEDDED_POSTGRESQL;
+import static org.springframework.core.Ordered.HIGHEST_PRECEDENCE;
+
+@Slf4j
+@Configuration
+@Order(HIGHEST_PRECEDENCE)
+@ConditionalOnProperty(name = "embedded.postgresql.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(PostgreSQLProperties.class)
+public class EmbeddedPostgreSQLBootstrapConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    PostgreSQLStatusCheck postgresSQLStartupCheckStrategy(PostgreSQLProperties properties) {
+        return new PostgreSQLStatusCheck(properties);
+    }
+
+    @Bean(name = BEAN_NAME_EMBEDDED_POSTGRESQL, destroyMethod = "stop")
+    public GenericContainer postgresql(ConfigurableEnvironment environment,
+                                       PostgreSQLProperties properties,
+                                       PostgreSQLStatusCheck postgreSQLStatusCheck) {
+        log.info("Starting postgresql server. Docker image: {}", properties.dockerImage);
+
+        GenericContainer postgresql =
+            new GenericContainer(properties.dockerImage)
+                .withEnv("POSTGRES_USER", properties.getUser())
+                .withEnv("POSTGRES_PASSWORD", properties.getPassword())
+                .withEnv("PGPASSWORD", properties.password) // for health check
+                .withEnv("POSTGRES_DB", properties.getDatabase())
+                .withCommand("postgres")
+                .withLogConsumer(containerLogsConsumer(log))
+                .withExposedPorts(properties.port)
```

Comment:
pick random free port by default?

Reply:
https://github.com/Playtika/testcontainers-spring-boot/search?utf8=✓&q=withExposedPorts&type=

Reply:
ok, see now

Reply:
Started postgresql server. Connection details: {embedded.postgresql.port=32899, embedded.postgresql.host=localhost, embedded.postgresql.schema=test_db, embedded.postgresql.user=postgresql, embedded.postgresql.password=letmein}, JDBC connection url: jdbc:postgresql://localhost:32899/test_db
public port is random